django_declarative_apis/authentication/oauthlib/oauth1.py

```diff
@@ -48,15 +48,15 @@ def validate_missing_parameters(self, request, parameters=None):
 
         params.extend(parameters)
 
-        collected_request_parameters = dict(
-            signature.collect_parameters(
-                uri_query=request.GET.urlencode(),
-                body=request.POST.dict(),
-                headers=request.META,
-                exclude_oauth_signature=False,
-            )
-        )
         try:
+            collected_request_parameters = dict(
```

Comment:
Does this raise a different exception type? I'm thinking that we won't be able to differentiate between missing auth headers and malformed ones anymore, something that I'm pretty sure we'll still want.

Reply:
Here's the path I went down here:

1. A "Malformed authorization header" currently raises a `ValueError` from `oauthlib` which is not caught anywhere: [oauthlib:parse_authorization_header](https://github.com/oauthlib/oauthlib/blob/7d7fe904af504c20f7d802650e54a78e3d0be8ba/oauthlib/oauth1/rfc5849/utils.py#L83)
2. DDA attempts to massage most OAuth errors into ["those specified in the OAuth Problem Reporting proposal"](
http://wiki.oauth.net/w/page/12238543/ProblemReporting), [here](https://github.com/salesforce/django-declarative-apis/blob/main/django_declarative_apis/authentication/oauthlib/oauth_errors.py#LL53C5-L54C55), which is what our service expects in the place this bubbles up
3. Looking through that list of errors, I figured `OAuthMissingParameterError` was still the best fit in this case.

Do we currently detect a missing Authorization header intelligently?

Reply:
In this case it's missing _all_ the needed parameters, and so the error lists them all.

Reply:
Right, so that would work in this case. However, if there's a case where we're getting an invalid auth header (anything that doesn't conform explicitly to `OAuth ` as defined [here](https://github.com/oauthlib/oauthlib/blob/7d7fe904af504c20f7d802650e54a78e3d0be8ba/oauthlib/oauth1/rfc5849/utils.py#L77)), it'll get swallowed up as a `missing` error. I think those two cases should be distinguished from one another.

Optimally, I think that oauthlib would report `oauth_parameters_rejected` (defined in [OAuth error reporting](http://wiki.oauth.net/w/page/12238543/ProblemReporting)) rather than simply a `ValueError` but it wasn't implemented that way. I think that the best thing to do would be to add support for [token_revoked](token_revoked) in [build_error](https://github.com/salesforce/django-declarative-apis/blob/main/django_declarative_apis/authentication/oauthlib/oauth_errors.py#L64) and handle that case when we get a `ValueError` from `collect_parameters`. That way, the caller can clearly see the difference between missing and invalid parameters.

We can also add an explicit check for all parameters missing case you mentioned here too in order to catch that case.

Reply:
After discussion in Slack - I'm returning `parameter_rejected`